tw-tkms-starter/src/test/java/com/transferwise/kafka/tkms/PostgresEndToEndIntTest.java

```diff
@@ -10,20 +12,28 @@
 public class PostgresEndToEndIntTest extends EndToEndIntTest {
 
   private static Stream<Arguments> compressionInput() {
-    return Stream.of(
-        Arguments.of(CompressionAlgorithm.GZIP, 111, 110),
-        Arguments.of(CompressionAlgorithm.NONE, 1171, 1171),
-        Arguments.of(CompressionAlgorithm.LZ4, 134, 134),
-        Arguments.of(CompressionAlgorithm.SNAPPY, 160, 160),
-        Arguments.of(CompressionAlgorithm.SNAPPY_FRAMED, 158, 158),
-        Arguments.of(CompressionAlgorithm.ZSTD, 100, 100)
-    );
+    var deferUntilCommits = List.of(false, true);
+    var arguments = new ArrayList<Arguments>();
+
+    for (var deferUntilCommit : deferUntilCommits) {
+      return Stream.of(
+          Arguments.of(CompressionAlgorithm.GZIP, 111, 110, deferUntilCommit),
+          Arguments.of(CompressionAlgorithm.NONE, 1171, 1171, deferUntilCommit),
+          Arguments.of(CompressionAlgorithm.LZ4, 134, 134, deferUntilCommit),
+          Arguments.of(CompressionAlgorithm.SNAPPY, 160, 160, deferUntilCommit),
+          Arguments.of(CompressionAlgorithm.SNAPPY_FRAMED, 158, 158, deferUntilCommit),
+          Arguments.of(CompressionAlgorithm.ZSTD, 100, 100, deferUntilCommit)
+      );
+    }
+
+    return arguments.stream();
```

Comment:
This looks unreachable

Reply:
Yep, bad copy paste. Interesting that ide did not show it for me.